tests/suites/test_suite_psa_crypto.function

```diff
@@ -9783,6 +9847,59 @@ exit:
 }
 /* END_CASE */
 
+#if defined MBEDTLS_THREADING_PTHREAD
+
+/* BEGIN_CASE depends_on:MBEDTLS_THREADING_PTHREAD */
+void concurrently_generate_keys(int type_arg,
```

Comment:
Not critical for 3.6.0, but I'd like to address this soon after: multithreaded tests can have problems of their own. It would be convenient to have them in their own test suite(s).

We would like to [split `test_suite_psa_crypto`](https://github.com/Mbed-TLS/mbedtls/issues/4967). It's already too big. So once the release is over, please move to a separate file.

If there are useful functions in the header section of `test_suite_psa_crypto.function`, they can move to `tests/src/` now. (Historically, part of the reason `test_suite_psa_crypto.function` is so big is that when it started, every test suite had to be self-contained, so all the PSA code had to be in one place. But that was a long time ago.)

Reply:
I'll make a follow-up PR for this when I have time. I also want to add some extra documentation to the test as part of the PR: this test does not support data which has more threads than there are key slots (due to resource exhaustion), I didn't consider this when writing the test; I'll make sure to guard and document this caveat.